File: OOP And Phthon Prog/Assignment 01/4_problem.py
def solve(arr):
    count = 0
    all_even = True

    for x in arr:
        if x % 2 != 0:
            all_even = False
            break

    if all_even:
        for i in range(len(arr)):
            arr[i] = arr[i] // 2
        count += 1

    return count

num = int(input())
arr = list(map(int, input().split()))

result = solve(arr)
print(result)

File: OOP And Phthon Prog/Assignment 01/4_problem.cpp
#include <bits/stdc++.h>
using namespace std;

int solve(int arr[], int num) {
    int count = 0;
    bool isEven = true;
    for (int i = 0; i < num; i++) {
        if (arr[i] % 2 != 0) {
            isEven = false;
            break;
        }
    }
    if (isEven) {
        for (int i = 0; i < num; i++) {
            arr[i] = arr[i] / 2;
        }
        count++;
    }

    return count;
}

int main() {
    int num;
    cin >> num;
    int arr[num];
    for (int i = 0; i < num; i++) {
        cin >> arr[i];
    }
    cout << solve(arr, num);
    return 0;
}
